### DISCUSSION TITLE
What is the reasoning behind setting fzf as the default picker?

### DISCUSSION THREAD

**Discussion by alexandersix** [LazyVim/LazyVim#5140] [Q&A] [answered]
Hi everyone! Apologies if this has been answered before–my google-fu is failing me here.

I was just curious what the rationale is behind changing the default picker from Telescope to fzf.

I'm not here to complain about the change (or be negative about it in any way) and I know I can set either as my picker and things will work just fine, but as a user of Telescope, I'm honestly just curious if I'm missing some benefit(s) of fzf over Telescope!

P.S., I've already started to use fzf, but definitely not enough for an in-depth comparison. So thankful for the `LazyExtras` making it simple to switch back and forth 🙌🏻

**Comment by agriffis** [comment]
I think the main reason is speed. The difference is pronounced in a large repo.

I think that's also the main reason for switching from nvim-cmp to blink.

But you're right, there's no mention of the reason for the switch at https://www.lazyvim.org/news

**Reply by alexandersix** [reply]
Yeah, that would make sense! If this is the official reason, now you're going to send me down a rabbit hole of trying to figure out why fzf has such a performance boost over Telescope 😅

**Reply by ibhagwan** [reply, marked as answer]
> Yeah, that would make sense! If this is the official reason, now you're going to send me down a rabbit hole of trying to figure out why fzf has such a performance boost over Telescope 😅

The difference is not easily experienced with simple pickers or even files picker (not enough entries usually), the difference comes into play when you have a **large amount of lines in the picker for fuzzy matching**.

If you want to experience the difference in very palpable way, download a huge project, something like [Linux kernel source](https://github.com/torvalds/linux) then run `:Telescope grep_string`, wait for all the lines of the project to get fed into the picker and try to fuzzy match anything, then compare that to `:FzfLua grep_project`.

Note this is just for the demonstration, the “right” way would be opening `live_grep`, reducing the result set to something more manageable and then pressing `ctrl-g` to switch to regular grep (fuzzy matching).

**Reply by alexandersix** [reply]
> > Yeah, that would make sense! If this is the official reason, now you're going to send me down a rabbit hole of trying to figure out why fzf has such a performance boost over Telescope 😅
> 
> The difference is not easily experienced with simple pickers or even files picker (not enough entries usually), the difference comes into play when you have a **large amount of lines in the picker for fuzzy matching**.
> 
> If you want to experience the difference in very palpable way, download a huge project, something like [Linux kernel source](https://github.com/torvalds/linux) then run `:Telescope grep_string`, wait for all the lines of the project to get fed into the picker and try to fuzzy match anything, then compare that to `:FzfLua grep_project`.
> 
> Note this is just for the demonstration, the “right” way would be opening `live_grep`, reducing the result set to something more manageable and then pressing `ctrl-g` to switch to regular grep (fuzzy matching).

That makes a ton of sense – thanks for the in-depth answer!

**Comment by moonc4ke** [comment]
Yeah, I would love to know the reason also :) anyway thanks a lot for the great work!

**Comment by mendesbarreto** [comment]
I tried to search for some reasons and could not find the why. Even the person who spoke about the performance could not say where exactly the boost happened; I am still curious about it...

**Reply by ibhagwan** [reply]
> I tried to search for some reasons and could not find the why. Even the person who spoke about the performance could not say where exactly the boost happened; I am still curious about it...

https://github.com/LazyVim/LazyVim/discussions/5140#discussioncomment-11720875

**Reply by mendesbarreto** [reply]
I read your answer but, my question was more related when they both used the same command-line fuzzy finder like FZF instead of the default algo of Telescope?

**Reply by folke** [reply]
- Open your home direcory (mine has 1.6million files).
- open Telescope (with fzf-native)
- wait till it's fully loaded
- start typing
- notice a long delay before the matching / sorting finishes

now do the same with fzf-lua.
profit
